components/brave_wallet/browser/zcash/zcash_keyring.cc

```diff
@@ -84,4 +184,28 @@ std::optional<std::vector<uint8_t>> ZCashKeyring::SignMessage(
   return hd_key->SignDer(message);
 }
 
+void ZCashKeyring::ConstructRootHDKey(const std::vector<uint8_t>& seed,
+                                      const std::string& hd_path) {
+  HDKeyring::ConstructRootHDKey(seed, hd_path);
+
+#if BUILDFLAG(ENABLE_ORCHARD)
+  if (!seed.empty() && IsZCashShieldedEnabled()) {
+    auto orchard_key = HDKeyZip32::GenerateFromSeed(seed);
+    if (!orchard_key) {
+      NOTREACHED();
+      return;
```

Comment:
this is an anti-pattern, see https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/checks.md#examples